werkit/compute/graph/_state_manager.py

```diff
@@ -47,18 +48,26 @@ def evaluate(
             else:
                 self._assert_known_keys(targets)
 
+        def wrap_node(name, node):
+            wrapped = node.bind(self.instance)
+
+            def wrapper(*args):
+                value = wrapped(*args)
+                return node.coerce(name, value)
+
+            functools.update_wrapper(wrapper, wrapped)
+            return wrapper
+
         afx = Artifax(
             {
-                name: self.dependency_graph.compute_nodes[name].bind(self.instance)
-                for name in self.dependency_graph.compute_nodes.keys()
+                name: wrap_node(name, node)
+                for name, node in self.dependency_graph.compute_nodes.items()
             }
         )
         if self.store:
             afx.set(**self.store)
         afx.build(targets=targets)
-        # TODO: `afx.build()` should always return an object.
```

Comment:
We're using an Artifax implementation detail to get the result so this todo (for the artifax API) is still relevant.